### ISSUE TITLE
API not behaving as documented

### ISSUE THREAD

**Issue by ledav78** [nilsteampassnet/TeamPass#1665]
Hello, I'm trying to use the API, but I receive answers which are not in line with the documentation.
I have the same results either with powershell or with curl.
Any help is welcome.

### Steps to reproduce
We have activated the API and generated and API key

Here below the results for some API calls:
**Return Api Version:**
`curl -k -X GET https://servername/teampass/api/index.php/info/version?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"err":"Method not authorized"}**

**Get complexity levels:**
`curl -k -X GET https://servername/teampass/api/index.php/info/complexicity_levels_list?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"err":"Method not authorized"}**

**Generate password :**
`curl -k -X GET https://servername/teampass/api/index.php/new_password/12;0;1;1;0;0;0?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"err":"Method not authorized"}**

**Read folder : folder ID 662 exists and contains items** 
`curl -k -X GET https://servername/teampass/api/index.php/read/folder/662?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"err":"No results"}**

**Read Item: this one seems to be working**
`curl -k -X GET https://servername/teampass/api/index.php/read/items/1716?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"1716":{"label":"Compte fournisseur","login":"**************","url":"","pw":"***********"}}**

**Write Item:**
Item definition is : TestAPI;T$est@p1;TestDecription;662;hopital\testapi;test@test.be;www.test.be;test;1
Which converted in base64 gives VGVzdEFQSTtUQHAxO1Rlc3REZWNyaXB0aW9uOzY2Mjtob3BpdGFsXHRlc3RhcGk7dGVzdEB0ZXN0LmJlO3d3dy50ZXN0LmJlO3Rlc3Q7MQ==

`curl -k -X GET https://servername/teampass/api/index.php/add/item/VGVzdEFQSTtUQHAxO1Rlc3REZWNyaXB0aW9uOzY2Mjtob3BpdGFsXHRlc3RhcGk7dGVzdEB0ZXN0LmJlO3d3dy50ZXN0LmJlO3Rlc3Q7MQ==?apikey=ooZ9zeesheeFeij5foh7oavee8aeHei9Aihu4fe`
**{"err":"Item definition not complete"}**

### Server configuration
**Operating system**: CentOS
**Web Server** : Apache 2.2.22, with https enabled
**Teampass version:** 2.1.26
**Updated from an older Teampass or fresh install:** Updated

### Client configuration
**Operating system:** Windows 7

**Comment by nilsteampassnet** [comment]
To what documentation source do you refer? ReadThedocs?

In this case, it is normal. It fits the 2.0 version which is delivered with TP 2.1.27.

Notice that API will evolve more in next release of Teampass.

**Comment by ledav78** [comment]
Hi Nils,
Indeed, by looking more deeply at the php code, I figured out today that the documentation on ReadTheDocs describe the API of TP 2.1.27.

Finaly, I was able to create a new item with the ";" separator form of the URL.
I still have issue to do it from powershell, but I'll probably find the solution.
In parallel, we will have a look at the upgrade.

Thanks for your feedback.
Topic can be closed.

David
